packages/ckeditor5-remove-format/src/removeformatcommand.ts

```diff
@@ -52,7 +54,18 @@ export default class RemoveFormatCommand extends Command {
 					const itemRange = writer.createRangeOn( item );
 
 					for ( const attributeName of this._getFormattingAttributes( item, schema ) ) {
-						writer.removeAttribute( attributeName, itemRange );
+						if ( isGHSAttributeName( attributeName ) ) {
```

Comment:
Maybe we could verify an attribute if it's a GHS by the `Schema#setAttributeProperties()` with some GHS dedicated property? `AttributeProperties` already allows for non-engine-defined properties: 
https://github.com/ckeditor/ckeditor5/blob/0c3ab39c5e211d131c865ef44b8ef872656960c0/packages/ckeditor5-engine/src/model/schema.ts#L2067

Or maybe it should use some API exposed from the GHS plugin? See other comments.

Reply:
I added `getGhsAttributesForElement` method to GHS plugin. Let's continue in this thread https://github.com/ckeditor/ckeditor5/pull/16867#discussion_r1706475443.